include/wil/chrono.h

```diff
@@ -0,0 +1,374 @@
+        }
+
+        inline __WI_LIBCPP_CONSTEXPR_AFTER_CXX11 cpu_time_duration get_cpu_time(const execution_times& times, cpu_time kind) WI_NOEXCEPT
+        {
+            switch (kind)
+            {
+                case cpu_time::total: return times.kernel_time + times.user_time;
+                case cpu_time::kernel: return times.kernel_time;
+                case cpu_time::user: return times.user_time;
+                DEFAULT_UNREACHABLE;
+            }
+        }
+    }
+
+    using thread_times = execution_times;
+
+    template<class error_policy>
+    WI_NODISCARD thread_times get_thread_times(HANDLE thread = ::GetCurrentThread())
+    {
+        FILETIME creation_time;
+        FILETIME exit_time;
+        FILETIME kernel_time;
+        FILETIME user_time;
+
+        error_policy::Win32BOOL(::GetThreadTimes(thread, &creation_time, &exit_time, &kernel_time, &user_time));
+
+        return details::execution_times_from_filetimes(creation_time, exit_time, kernel_time, user_time);
+    }
+
+    WI_NODISCARD inline thread_times get_thread_times(HANDLE thread = ::GetCurrentThread())
+    {
+        return get_thread_times<err_exception_policy>(thread);
+    }
+
+    template<class error_policy>
+    WI_NODISCARD cpu_time_duration get_thread_cpu_time(HANDLE thread = ::GetCurrentThread(), cpu_time kind = cpu_time::total)
+    {
+        return details::get_cpu_time(get_thread_times<error_policy>(thread), kind);
+    }
+
+    WI_NODISCARD inline cpu_time_duration get_thread_cpu_time(HANDLE thread = ::GetCurrentThread(), cpu_time kind = cpu_time::total)
+    {
+        return get_thread_cpu_time<err_exception_policy>(thread, kind);
+    }
+
+    using process_times = execution_times;
+
+    template<class error_policy>
+    WI_NODISCARD process_times get_process_times(HANDLE process = ::GetCurrentProcess())
+    {
+        FILETIME creation_time;
+        FILETIME exit_time;
+        FILETIME kernel_time;
+        FILETIME user_time;
+
+        error_policy::Win32BOOL(::GetProcessTimes(process, &creation_time, &exit_time, &kernel_time, &user_time));
+
+        return details::execution_times_from_filetimes(creation_time, exit_time, kernel_time, user_time);
+    }
+
+    WI_NODISCARD inline process_times get_process_times(HANDLE process = ::GetCurrentProcess())
+    {
+        return get_process_times<err_exception_policy>(process);
+    }
+
+    template<class error_policy>
+    WI_NODISCARD cpu_time_duration get_process_cpu_time(HANDLE process = ::GetCurrentProcess(), cpu_time kind = cpu_time::total)
+    {
+        return details::get_cpu_time(get_process_times<error_policy>(process), kind);
+    }
+
+    WI_NODISCARD inline cpu_time_duration get_process_cpu_time(HANDLE process = ::GetCurrentProcess(), cpu_time kind = cpu_time::total)
+    {
+        return get_process_cpu_time<err_exception_policy>(process, kind);
+    }
+#pragma endregion
+
+#pragma region Thread/process CPU clocks
+    struct current_thread_cpu_time_clock

```

Comment:
Requires `is_steady` static member to satisfy [Clock](https://en.cppreference.com/w/cpp/named_req/Clock) requirements

Reply:
Same with `current_process_cpu_time_clock` and possibly other I may have missed